redis/src/commands/json.rs

```diff
@@ -0,0 +1,218 @@
+// can't use rustfmt here because it screws up the file.
+#![cfg_attr(rustfmt, rustfmt_skip)]
+use crate::cmd::{cmd, Cmd};
+use crate::connection::ConnectionLike;
+use crate::pipeline::Pipeline;
+use crate::types::{FromRedisValue, RedisResult, ToRedisArgs};
+use crate::{RedisError, ErrorKind};
+
+#[cfg(feature = "cluster")]
+use crate::commands::ClusterPipeline;
+
+use serde::ser::Serialize;
+
+impl From<serde_json::Error> for RedisError {
+	fn from(serde_err: serde_json::Error) -> RedisError {
+		RedisError::from((
+			ErrorKind::Serialize,
+			"Serialization Error",
+			format!("{}", serde_err)
+		))
+	}
+}
+
+implement_json_commands! {
+    'a
+	
+	/// Append the JSON `value` to the array at `path` after the last element in it.
+	fn json_arr_append<K: ToRedisArgs, P: ToRedisArgs, V: Serialize>(key: K, path: P, value: &'a V) {
+		Ok::<Cmd, RedisError>(
```

Comment:
If you rewrite it like this:

```rust
let mut cmd = cmd("JSON.ARRAPPEND");
cmd.arg(key)
    .arg(path)
    .arg(serde_json::to_string(value)?);
Ok::<_, RedisError>(cmd)
```

I think you no longer need `arg_take()`? (Also note that we don't need to specify the `Cmd` type in the `Ok` turbofish.)

Also while looking this I noticed all the indentation in this file is tabs instead of spaces, should fix that up (VS Code has a `Convert Indentation to Spaces` thing.)

Reply:
I'll do some investigation into this, I think I tried something like this, but I may be wrong

Reply:
I tried it for one command and it seemed to work? But maybe I got it wrong...

Reply:
> I tried it for one command and it seemed to work? But maybe I got it wrong...

sorry, I've been busy with a hackathon over the weekend, will look into this asap

Reply:
No hurry from my side! Hope you had fun at the hackathon.

Reply:
> If you rewrite it like this:
> 
> ```rust
> let mut cmd = cmd("JSON.ARRAPPEND");
> cmd.arg(key)
>     .arg(path)
>     .arg(serde_json::to_string(value)?);
> Ok::<_, RedisError>(cmd)
> ```
> 
> I think you no longer need `arg_take()`? (Also note that we don't need to specify the `Cmd` type in the `Ok` turbofish.)
> 
> Also while looking this I noticed all the indentation in this file is tabs instead of spaces, should fix that up (VS Code has a `Convert Indentation to Spaces` thing.)

```rust
let mut cmd = cmd("JSON.ARRAPPEND");

cmd.arg(key)
	.arg(path)
	.arg(serde_json::to_string(value)?);

Ok::<_, RedisError>(cmd)
```

Does indeed seem to work, ill update the rest of them

I've run the command to use spaces instead of tabs